chain/penumbra/penumbra_app_node.go

```diff
@@ -132,47 +193,69 @@ func (p *PenumbraAppNode) GenerateGenesisFile(
 	if err != nil {
 		return fmt.Errorf("error marshalling validators to json: %w", err)
 	}
+
 	fw := dockerutil.NewFileWriter(p.log, p.DockerClient, p.TestName)
 	if err := fw.WriteFile(ctx, p.VolumeName, "validators.json", validatorsJson); err != nil {
 		return fmt.Errorf("error writing validators to file: %w", err)
 	}
-	allocationsCsv := []byte(`"amount","denom","address"\n`)
+
+	allocationsCsv := []byte("\"amount\",\"denom\",\"address\"\n")
 	for _, allocation := range allocations {
-		allocationsCsv = append(allocationsCsv, []byte(fmt.Sprintf(`"%s","%s","%s"\n`, allocation.Amount.String(), allocation.Denom, allocation.Address))...)
+		allocationsCsv = append(allocationsCsv, []byte(fmt.Sprintf("\"%s\",\"%s\",\"%s\"\n", allocation.Amount.String(), allocation.Denom, allocation.Address))...)
 	}
+
 	if err := fw.WriteFile(ctx, p.VolumeName, "allocations.csv", allocationsCsv); err != nil {
 		return fmt.Errorf("error writing allocations to file: %w", err)
 	}
+
 	cmd := []string{
 		"pd",
 		"testnet",
 		"generate",
 		"--chain-id", chainID,
+		"--preserve-chain-id",
 		"--validators-input-file", p.ValidatorsInputFileContainer(),
 		"--allocations-input-file", p.AllocationsInputFileContainer(),
 	}
 	_, _, err = p.Exec(ctx, cmd, nil)
+	if err != nil {
+		return fmt.Errorf("failed to exec testnet generate: %w", err)
+	}
+
 	return err
 }
 
 func (p *PenumbraAppNode) GetAddress(ctx context.Context, keyName string) ([]byte, error) {
-	cmd := []string{"pcli", "-d", p.HomeDir(), "addr", "list"}
+	keyPath := filepath.Join(p.HomeDir(), "keys", keyName)
+	pdUrl := fmt.Sprintf("http://%s:8080", p.HostName())
+	cmd := []string{"pcli", "-d", keyPath, "-n", pdUrl, "view", "address"}
+
 	stdout, _, err := p.Exec(ctx, cmd, nil)
 	if err != nil {
 		return nil, err
 	}
-	addresses := strings.Split(string(stdout), "\n")
-	for _, address := range addresses {
-		fields := strings.Fields(address)
-		if len(fields) < 3 {
-			continue
-		}
-		if fields[1] == keyName {
-			// TODO penumbra address is bech32m. need to decode to bytes here
-			return []byte(fields[2]), nil
-		}
+
+	if len(stdout) == 0 {
+		return []byte{}, errors.New("address not found")
 	}
-	return []byte{}, errors.New("address not found")
+
+	addr := strings.TrimSpace(string(stdout))
+	return []byte(addr), nil
+}
+
+// TODO we need to change the func sig to take a denom then filter out the target denom bal from stdout
+func (p *PenumbraAppNode) GetBalance(ctx context.Context, keyName string) (int64, error) {
+	keyPath := filepath.Join(p.HomeDir(), "keys", keyName)
+	pdUrl := fmt.Sprintf("http://%s:8080", p.HostName())
+	cmd := []string{"pcli", "-d", keyPath, "-n", pdUrl, "view", "balance"}
+
+	stdout, _, err := p.Exec(ctx, cmd, nil)
+	if err != nil {
+		return 0, err
+	}
+
+	fmt.Printf("STDOUT BAL: '%s'\n", string(stdout))
+	return 0, nil
```

Comment:
I'm not 100% sure, they do technically return the same information the only difference being which piece of software is used to retrieve the balance (`pclientd` vs. `pcli`). 

This PR moves away from making queries and broadcasting txs from the `PenumbraAppNode` in favor of the `PenumbraClientNode` when possible, so technically this method is unused now within `interchaintest` but it's not immediately clear to me if it will be used moving forward or if it's beneficial to expose methods on both objects. 

I could remove this for now, change the method signature to accept a denom string and actually process the output properly so we can remove the todo, or open up an issue to come back to this later in a follow up PR. Thoughts?